### PR TITLE
Fix reflection warnings; kondo config & parallel test improvements

### PR DESCRIPTION
**Pull request by camsaul**
Misc improvements:

- Fix reflection warnings in tests 
- Configure Kondo to disallow `with-redefs` and `eval` in non-test code (not really an issue but an ounce of prevention is worth a pound of cure or whatever), and to disallow `clojure.pprint/pprint` the same way we disallow `println` 
- Mark some more tests as `^:parallel`

### CODE REVIEW

File: .clj-kondo/hooks/clojure/test.clj [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+(def disallowed-parallel-forms
+  '#{with-redefs
+     clojure.core/with-redefs
+     metabase.test/with-temporary-setting-values
+     mt/with-temporary-setting-values})
+
+(defn warn-about-disallowed-parallel-forms [form]
+  (letfn [(f [form]
+            (when (hooks/token-node? form)
+              (let [sexpr (hooks/sexpr form)]
```
Comment (metamben) [inline .clj-kondo/hooks/clojure/test.clj:13]: Would
```suggestion
            (when-let [sexpr (and (hooks/token-node? form)
                                  (hooks/sexpr form))]
```
 be an obscure misuse?

File: .clj-kondo/hooks/clojure/test.clj [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+                (when (symbol? sexpr)
+                  (when (or (disallowed-parallel-forms sexpr)
+                            #_(= (last (str sexpr)) \!))
```
Comment (metamben) [inline .clj-kondo/hooks/clojure/test.clj:16]:
```suggestion
                (when (disallowed-parallel-forms sexpr))
```

File: .clj-kondo/hooks/clojure/test.clj [hunk 3]
```diff
@@ -0,0 +1,46 @@
+(ns hooks.clojure.test
+  (:require [clj-kondo.hooks-api :as hooks]))
+
+(def disallowed-parallel-forms
+  '#{with-redefs
+     clojure.core/with-redefs
+     metabase.test/with-temporary-setting-values
+     mt/with-temporary-setting-values})
```
Comment (metamben) [inline .clj-kondo/hooks/clojure/test.clj:8]: What about `alter-var-root` and functions using that like `driver/register!`?

File: test/metabase/driver_test.clj [hunk 4]
```diff
@@ -41,7 +44,7 @@
     (is (driver/available? "metabase.driver-test/test-driver")
         "`driver/available?` should work for if `driver` is a string -- see #10135")))
 
-(deftest unique-connection-property-test
+(deftest ^:parallel unique-connection-property-test
```
Comment (metamben) [inline test/metabase/driver_test.clj:47]: This means `tx.env/set-test-drivers!` should also trigger warnings in `:parallel` tests, right?

File: test/metabase/models/secret_test.clj [hunk 5]
```diff
@@ -184,4 +184,4 @@
             (let [decoded (decode-ssl-db-property content mime-type property)]
               (is (instance? (Class/forName "[B") decoded))
```
Comment (metamben) [inline test/metabase/models/secret_test.clj:185]:
```suggestion
              (is (bytes? decoded))
```

**Review by metamben — APPROVED** [review]
👍 

You don't mention that model invocations are allowed again, but I suppose this is an intended change.

**Comment by camsaul** [comment]
> You don't mention that model invocations are allowed again, but I suppose this is an intended change.

They're not, but now that we're using Toucan 2 if you try them they won't work, so it's not really something we need to enforce with Kondo anymore. The Kondo rule was only in there because with Toucan 1 they actually did do something

File: .clj-kondo/hooks/clojure/test.clj (hunk 3 above)
Comment (camsaul) [reply .clj-kondo/hooks/clojure/test.clj:8]: Yeah, this should have wayyy more things in it than just these. But I ran into https://github.com/clj-kondo/clj-kondo/issues/1996 while working on this so I thought I'd just include what I had and pick this back up once that issue was fixed upstream

File: test/metabase/driver_test.clj (hunk 4 above)
Comment (camsaul) [reply test/metabase/driver_test.clj:47]: Yes, but I'll do that as a follow-on

**Pull request merged**